ddl/backfilling_dispatcher.go

```diff
@@ -55,54 +71,58 @@ func (*backfillingDispatcherExt) OnTick(_ context.Context, _ *proto.Task) {
 }
 
 // OnNextSubtasksBatch generate batch of next stage's plan.
-func (h *backfillingDispatcherExt) OnNextSubtasksBatch(ctx context.Context, _ dispatcher.TaskHandle, gTask *proto.Task) ([][]byte, error) {
-	var globalTaskMeta BackfillGlobalMeta
-	if err := json.Unmarshal(gTask.Meta, &globalTaskMeta); err != nil {
+func (h *backfillingDispatcherExt) OnNextSubtasksBatch(ctx context.Context,
+	taskHandle dispatcher.TaskHandle,
+	gTask *proto.Task,
+) (taskMeta [][]byte, err error) {
+	var gTaskMeta BackfillGlobalMeta
+	if err := json.Unmarshal(gTask.Meta, &gTaskMeta); err != nil {
 		return nil, err
 	}
-	job := &globalTaskMeta.Job
-	var tblInfo *model.TableInfo
+
+	useExtStore := len(gTaskMeta.CloudStorageURI) > 0
+	defer func() {
+		// Only redact when the task is complete.
+		if len(taskMeta) == 0 && useExtStore {
+			redactCloudStorageURI(ctx, gTask, &gTaskMeta)
+		}
+	}()
+
+	job := &gTaskMeta.Job
 	tblInfo, err := getTblInfo(h.d, job)
 	if err != nil {
 		return nil, err
 	}
-	var subTaskMetas [][]byte
 	// generate partition table's plan.
 	if tblInfo.Partition != nil {
-		if gTask.Step != proto.StepInit {
-			// This flow for partition table has only one step
+		switch gTask.Step {
+		case stageInit:
+			gTask.Step = stageReadIndex
```

Comment:
framework also changes step, we need unify where we can change steps, see https://github.com/pingcap/tidb/pull/46593/files#diff-2a828989b540648a0bdef3d984d7946ea4fcb3c18718a4047b320c712ee1b855R424

Reply:
> framework also changes step, we need unify where we can change steps, see https://github.com/pingcap/tidb/pull/46593/files#diff-2a828989b540648a0bdef3d984d7946ea4fcb3c18718a4047b320c712ee1b855R424

I'm trying to change it.